include/boost/asio/windows_sspi/detail/sspi_functions.hpp

```diff
@@ -0,0 +1,127 @@
+//
+// windows_sspi/detail/sspi_functions.hpp
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+//
+// Copyright (c) 2020 Kasper Laudrup (laudrup at stacktrace dot dk)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+
+#ifndef BOOST_ASIO_WINDOWS_SSPI_DETAIL_SSPI_FUNCTIONS_HPP
+#define BOOST_ASIO_WINDOWS_SSPI_DETAIL_SSPI_FUNCTIONS_HPP
+
+// TODO: Avoid cluttering global namespace (and avoid Windows headers if possible)
+#define SECURITY_WIN32
+#include <security.h>
+
+namespace {
+
+SecurityFunctionTable* function_table() {
+  static SecurityFunctionTable* impl = InitSecurityInterface();
+  // TODO: Figure out some way to signal this to the user instead of aborting
+  BOOST_ASSERT_MSG(impl != nullptr, "Unable to initialize SecurityFunctionTable");
+  return impl;
+}
+
+} // namespace
+
+namespace boost {
+namespace asio {
+namespace windows_sspi {
+namespace detail {
+namespace sspi_functions {
+
+SECURITY_STATUS AcquireCredentialsHandleA(LPSTR pszPrincipal,
```

Comment:
Free-functions defined in headers need to be `inline` to avoid ODR issues.

Reply:
Indeed they should. This is kind of embarrassing :-(